server/graphql/v2/mutation/TierMutations.js

```diff
@@ -40,14 +40,22 @@ const tierMutations = {
         args.tier.minimumAmount = null;
       }
 
-      return await tier.update({
+      const tierUpdateData = {
         ...args.tier,
         id: tierId,
         amount: getValueInCentsFromAmountInput(args.tier.amount),
         minimumAmount: args.tier.minimumAmount ? getValueInCentsFromAmountInput(args.tier.minimumAmount) : null,
         goal: args.tier.goal ? getValueInCentsFromAmountInput(args.tier.goal) : null,
         interval: getIntervalFromTierFrequency(args.tier.frequency),
-      });
+      };
+      let data;
+      if (args.tier.singleTicket !== undefined) {
+        const tier = await models.Tier.findOne({ where: { id: tierId } });
+        data = { ...tier.data, singleTicket: args.tier.singleTicket };
+        tierUpdateData.data = data;
+      }
+
+      return await tier.update(tierUpdateData);
```

Comment:
Why reload the tier here? Can't we just do

```suggestion

      if (args.tier.singleTicket !== undefined) {
        tierUpdateData.data = { ...tier.data, singleTicket: args.tier.singleTicket };
      }

      return tier.update(tierUpdateData);
```

Reply:
`tier.data` is always undefined here which is why I obtained it using the modal. 🤔

Reply:
Would be great to identify the root cause. If we can't fix it easily, then at least add a comment to explain while we're doing that; otherwise, the next engineer looking at it will rightfully refactor to simplify not knowing about the side-effect.

Note that @vaughanj10 [spotted](https://opencollective.slack.com/archives/C0471HKNW7J/p1665696797472689) some strange sequelize behavior with sequelize between `update` and `JSONB`  recently, it could be related.

Reply:
@Betree : Sorry, I was testing this again and I think I was mixing this up with the other code below (https://github.com/opencollective/opencollective-api/pull/8100#discussion_r1013417226) and thought this also needs the `tier.data` reload. But here it seems that it works find with the simplification. I've changed the code to what you've suggested. One thing I noted though is that there's some issues with the new tier page. For example, `TierUpdateInput` was missing the `invoiceTemplate` field so I added it,

https://github.com/opencollective/opencollective-api/blob/9205842941eed704bb84a0535e121f16a2bcc3a9/server/graphql/v2/input/TierUpdateInput.ts#L59-L61